### PR TITLE
feat(Dialog): Migrate to Variance + ColorModes

### PR DESCRIPTION
**Pull request by codecaaron**
### Overview

This is a no op change to Dialogs that switches the API to variance and removes unnecessary nodes.  Further refactor comes with the [Box Migration](https://github.com/Codecademy/client-modules/pull/1658)

* Updates Dialog to current color mode standard and removes extra code.
* Updates the documentation to reflect the differences.
* Simplifies internal CSS.

### PR Checklist

- [ ] Related to designs:
- [ ] Related to JIRA ticket: ABC-123
- [ ] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.10.3-alpha.62fe34.0<br/>@codecademy/gamut-kit@0.3.39-alpha.62fe34.0<br/>@codecademy/gamut-labs@14.0.1-alpha.62fe34.0<br/>@codecademy/gamut-styles@9.5.4-alpha.62fe34.0<br/>@codecademy/gamut-tests@2.3.15-alpha.62fe34.0<br/>@codecademy/gamut@32.0.1-alpha.62fe34.0<br/>@codecademy/konamimojisplosion@0.2.15-alpha.62fe34.0<br/>@codecademy/markdown-overrides@0.5.27-alpha.62fe34.0<br/>@codecademy/styleguide@36.0.1-alpha.62fe34.0<br/>@codecademy/variance@0.9.1-alpha.62fe34.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6080166baeb414fef6838325--gamut-preview.netlify.app>https://6080166baeb414fef6838325--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6080166baeb414fef6838325>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.10.5-alpha.8a618d.0<br/>@codecademy/gamut-kit@0.3.45-alpha.8a618d.0<br/>@codecademy/gamut-labs@14.2.5-alpha.8a618d.0<br/>@codecademy/gamut-styles@9.5.6-alpha.8a618d.0<br/>@codecademy/gamut-tests@2.3.17-alpha.8a618d.0<br/>@codecademy/gamut@32.1.4-alpha.8a618d.0<br/>@codecademy/konamimojisplosion@0.2.17-alpha.8a618d.0<br/>@codecademy/markdown-overrides@0.5.29-alpha.8a618d.0<br/>@codecademy/styleguide@36.2.6-alpha.8a618d.0<br/>@codecademy/variance@0.9.1-alpha.8a618d.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://608371ffcd53b334aaa40c41--gamut-preview.netlify.app>https://608371ffcd53b334aaa40c41--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/608371ffcd53b334aaa40c41>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.49-alpha.869efc.0<br/>@codecademy/gamut-labs@14.4.1-alpha.869efc.0<br/>@codecademy/gamut@32.3.1-alpha.869efc.0<br/>@codecademy/styleguide@36.4.1-alpha.869efc.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6086e7ba7a03e825bbb4b08c--gamut-preview.netlify.app>https://6086e7ba7a03e825bbb4b08c--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6086e7ba7a03e825bbb4b08c>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.10.7-alpha.07974d.0<br/>@codecademy/gamut-kit@0.3.51-alpha.07974d.0<br/>@codecademy/gamut-labs@14.4.3-alpha.07974d.0<br/>@codecademy/gamut-styles@9.6.1-alpha.07974d.0<br/>@codecademy/gamut-tests@2.3.19-alpha.07974d.0<br/>@codecademy/gamut@32.3.3-alpha.07974d.0<br/>@codecademy/konamimojisplosion@0.2.19-alpha.07974d.0<br/>@codecademy/markdown-overrides@0.5.31-alpha.07974d.0<br/>@codecademy/styleguide@36.4.3-alpha.07974d.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60874fbd8ba1342e464ee429--gamut-preview.netlify.app>https://60874fbd8ba1342e464ee429--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60874fbd8ba1342e464ee429>Deploy Logs</a></p>

File: packages/gamut/src/Dialog/index.tsx [hunk 1]
```diff
@@ -1,93 +1,57 @@
...
-      borderColor: 'navy',
-    },
-    dark: {
-      textColor: 'white',
-      backgroundColor: 'navy',
-      borderColor: 'white',
-    },
-  },
-});
+const ShroudedOverlay = styled(Overlay)(system.css({ bg: 'shadow' }));
```
Comment (dreamwasp) [inline packages/gamut/src/Dialog/index.tsx:13]: 😲

File: packages/styleguide/stories/Molecules/Dialog.stories.mdx [hunk 2]
```diff
@@ -60,7 +61,8 @@ import { PropsTable } from '~styleguide/blocks';
 
 ## Color Modes
 
-We currently support 2 color modes for Dialog `dark` and `light`.
+Dialog components respond to the current <LinkTo id="foundations-color-mode--page">Color Mode</LinkTo> they are used in. Each component inside of dialog will correctly display for the current color mode without any extra configuration.
+Check it out:
```
Comment (dreamwasp) [inline packages/styleguide/stories/Molecules/Dialog.stories.mdx:65]: 🔥

**Review by dreamwasp — APPROVED** [review]

**Review by JoshuaKGoldberg — APPROVED** [review]
Very cool. 🔥

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.63-alpha.281bc1.0<br/>@codecademy/gamut-labs@15.2.3-alpha.281bc1.0<br/>@codecademy/gamut@32.6.2-alpha.281bc1.0<br/>@codecademy/styleguide@36.8.2-alpha.281bc1.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://608b1aa58957236100e6f4e9--gamut-preview.netlify.app>https://608b1aa58957236100e6f4e9--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/608b1aa58957236100e6f4e9>Deploy Logs</a></p>

**Pull request merged**